openff/interchange/interop/openmm/_topology.py

```diff
@@ -15,9 +15,22 @@
 
 def to_openmm_topology(
     interchange: "Interchange",
+    collate: bool = False,
     ensure_unique_atom_names: str | bool = "residues",
 ) -> "openmm.app.Topology":
-    """Create an OpenMM Topology containing some virtual site information (if appropriate)."""
+    """
+    Create an OpenMM Topology containing some virtual site information (if appropriate).
+
+    Parameters
+    ----------
+    interchange
+        The Interchange object to convert to an OpenMM Topology.
+    collate
+        If False, the default, all virtual sites will be added to a single residue at the end of the topology.
```

Comment:
> single residue

I'm sure I'm missing something important here - is there a practical advantage to doing a single residue rather than a residue per molecule's worth of virtual sites?

Reply:
> upper bound on residue numbers

I could see AMBER and GROMACS being issues here :/ (at least iirc AMBER is still technically fixed width in parm7).

Reply:
Updated with this idea in mind:

```python
In [30]: x = (
    ...:     ForceField("opc.offxml")
    ...:     .create_interchange(
    ...:         Topology.from_molecules(
    ...:             [
    ...:                 Molecule.from_smiles("O"),
    ...:                 Molecule.from_smiles("O"),
    ...:             ],
    ...:         ),
    ...:     )
    ...:     .to_openmm_topology()
    ...: )

In [31]: for residue in x.residues():
    ...:     print(residue.index, [*residue.atoms()])
0 [<Atom 0 (O1x) of chain 0 residue 0 (UNK)>, <Atom 1 (H1x) of chain 0 residue 0 (UNK)>, <Atom 2 (H2x) of chain 0 residue 0 (UNK)>]
1 [<Atom 3 (O1x) of chain 1 residue 1 (UNK)>, <Atom 4 (H1x) of chain 1 residue 1 (UNK)>, <Atom 5 (H2x) of chain 1 residue 1 (UNK)>]
2 [<Atom 6 (EP) of chain 2 residue 2 (VS)>]
3 [<Atom 7 (EP) of chain 2 residue 3 (VS)>]
```

The only thing to note being that each virtual site has its own residue (which would have multiple virtual sites for a molecule that has more than one virtual site - for example, two TIP5P waters would have 3, 3, 2, 2 "atoms" in each "residue")

Reply:
I think that breakdown is great! That way you can quickly iterate through your residues and find the matching virtual sites.